lib/Exception/DocumentReferencedException.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/*
+ * This file is part of Sulu.
+ *
+ * (c) MASSIVE ART WebServices GmbH
+ *
+ * This source file is subject to the MIT license that is bundled
+ * with this source code in the file LICENSE.
+ */
+
+namespace Sulu\Component\DocumentManager\Exception;
+
+use PHPCR\PropertyInterface;
+
+/**
+ * This is exception is thrown if a document is still referenced, and the executed action does not work because of that.
+ */
+class DocumentReferencedException extends DocumentManagerException
+{
+    /**
+     * @var object
+     */
+    private $document;
+
+    /**
+     * @var PropertyInterface[]
+     */
+    private $references;
+
+    /**
+     * @param object $document
+     * @param PropertyInterface[] $references
+     * @param \Exception $previous
+     */
+    public function __construct($document, array $references = [], \Exception $previous = null)
+    {
+        $this->document = $document;
+        $this->references = $references;
+
+        parent::__construct(
+            sprintf(
+                'The document with the object hash "%s" cannot be deleted because it is still referenced',
```

Comment:
I know, but that is what is done in this library. You can still throw a typed exception, but I just prefer that the message is inline. I have seen instances where people just throw an existing exception that "looks like the right one" but gives a misleading error message. The message is the single most important thing of an Exception most of the time, it explains what has gone wrong and good exceptions also tell the developer how to fix the issue.